### PR TITLE
Port to Gnome 45: initial efforts

### PR DESCRIPTION
**Pull request by Sierra410**
After consulting with people in [#extensions:gnome.org](https://matrix.to/#/#extensions:gnome.org), it appears that the solution to backwards compatibility, is not doing backwards compatibility.  
Users will fetch the latest version of the extension their shell supports, so, codebases for 45+ and <44 should live in two different branches and be maintained, packaged, and uploaded separately.

The changes are not thoroughly tested as of now, so there might be some mistakes or things I've overlooked.

### CODE REVIEW

**Comment by raujonas** [comment]
Thanks for your help, that's much appreciated! I'll test it soon.

File: extension.js [hunk 1]
```diff
@@ -1,363 +1,344 @@
...
 
 const POSITIONS = {
     0: 'left',
     1: 'center',
-    2: 'right'
+    2: 'right',
 };
 
-function init() {
-    ExtensionUtils.initTranslations(Me.metadata.uuid);
```
Comment (raujonas) [inline extension.js:30]: Is there a way to enable the translations as well? It seems like it was completely removed or I'm looking in the wrong place.

**Review by raujonas — COMMENTED** [review]
Great job, it works. I only have two minor questions that I added as comments. I have fixed a null pointer error locally that was most likely there before, I'll commit that after merging, but otherwise it works perfectly well, thanks so much!

File: extension.js (hunk 1 above)
Comment (Sierra410) [reply extension.js:30]: From https://gjs.guide/extensions/upgrading/gnome-shell-45.html#extensionutils  

> `initTranslations()` Consider this method deprecated. Only specify gettext-domain in metadata.json. GNOME Shell can automatically initiate the translation for you when it sees the gettext-domain key in metadata.json.  

I assume it should just work. Should be verified through testing, of course.

File: prefs.js [hunk 2]
```diff
@@ -1,196 +1,192 @@
...
-    0: _('Left'),
-    1: _('Center'),
-    2: _('Right')
+    2: 'right',
 };
 
-function init() {
-    ExtensionUtils.initTranslations(Me.metadata.uuid);
-}
+export default class ExamplePreferences extends ExtensionPreferences {
```
[the remark this answers is not in the document]
Comment (Sierra410) [reply prefs.js:12]: Good catch! Overlooked this while copy-pasting boilerplate.

**Comment by Sierra410** [comment]
Another thing regarding this PR:  
Since almost every single line of the extension has been affected, I took the liberty to also run it through prettier using the [recommended settings](https://gjs.guide/guides/gjs/style-guide.html#prettier). Max number of columns doesn't seem to be specified, so I used 120 as a reasonable value.

**Comment by raujonas** [comment]
Thanks, that's a good idea. I should probably add that config file to the repo in the future, correct?

**Comment by Sierra410** [comment]
That could be done, yes.  
Honestly, I probably should've used ESlint, as that is what's preferred, according to gjs.guide.  
The extensions, however, aren't required to follow the style guide to the letter, so it's a bit of a persona preference, I supposed.

**Pull request merged**